eZ/Publish/Core/Persistence/Legacy/URL/Gateway/DoctrineDatabase.php

```diff
@@ -0,0 +1,259 @@
+<?php
+
+namespace eZ\Publish\Core\Persistence\Legacy\URL\Gateway;
+
+use eZ\Publish\Core\Persistence\Database\SelectQuery;
+use eZ\Publish\API\Repository\Values\URL\Query\Criterion;
+use eZ\Publish\API\Repository\Values\URL\Query\SortClause;
+use eZ\Publish\Core\Persistence\Legacy\URL\Gateway;
+use eZ\Publish\Core\Persistence\Legacy\URL\Query\CriteriaConverter;
+use eZ\Publish\Core\Persistence\Database\DatabaseHandler;
+use eZ\Publish\SPI\Persistence\URL\URL;
+use PDO;
+
+/**
+ * URL gateway implementation using the Doctrine.
+ */
+class DoctrineDatabase extends Gateway
+{
+    const URL_TABLE = 'ezurl';
+    const URL_LINK_TABLE = 'ezurl_object_link';
+
+    const COLUMN_ID = 'id';
+    const COLUMN_URL = 'url';
+    const COLUMN_ORIGINAL_URL_MD5 = 'original_url_md5';
+    const COLUMN_IS_VALID = 'is_valid';
+    const COLUMN_LAST_CHECKED = 'last_checked';
+    const COLUMN_MODIFIED = 'modified';
+    const COLUMN_CREATED = 'created';
+
+    /**
+     * Database handler.
+     *
+     * @param \eZ\Publish\Core\Persistence\Database\DatabaseHandler $dbHandler
+     */
+    protected $handler;
+
+    /**
+     * Criteria converter.
+     *
+     * @var \eZ\Publish\Core\Persistence\Legacy\URL\Query\CriteriaConverter
+     */
+    protected $criteriaConverter;
+
+    /**
+     * Creates a new Doctrine database Section Gateway.
+     *
+     * @param \eZ\Publish\Core\Persistence\Database\DatabaseHandler $dbHandler
```

Comment:
Please avoid `DatabaseHandler`. AFAIR this layer is supposed to be deprecated and removed in the future in favor of direct usage of `\Doctrine\DBAL\Connection`. // cc @andrerom 
This of course implies changes to entire Gateway implementation... Sorry! :(

Reply:
We will postpone this refactoring after a merge.